evaluate/task/task-write-test_test.go

```diff
@@ -216,3 +217,114 @@ func TestTaskWriteTestsRun(t *testing.T) {
 		})
 	})
 }
+
+func TestValidateWriteTestsRepository(t *testing.T) {
+	type testCase struct {
+		Name string
+
+		Before func(repositoryPath string)
+
+		RepositoryPath string
+		Language       language.Language
+
+		ExpectedErr string
+	}
+
+	validate := func(t *testing.T, tc *testCase) {
+		t.Run(tc.Name, func(t *testing.T) {
+			logOutput, logger := log.Buffer()
+			defer func() {
+				if t.Failed() {
+					t.Logf("Logging output: %s", logOutput.String())
+				}
+			}()
+
+			temporaryDirectoryPath := t.TempDir()
+			repositoryPath := filepath.Join(temporaryDirectoryPath, "testdata", tc.RepositoryPath)
+			require.NoError(t, osutil.CopyTree(filepath.Join("..", "..", "testdata", tc.RepositoryPath), repositoryPath))
+
+			if tc.Before != nil {
+				tc.Before(repositoryPath)
+			}
+
+			actualErr := validateWriteTestsRepository(logger, repositoryPath, tc.Language)
+			if len(tc.ExpectedErr) > 0 {
+				require.ErrorContains(t, actualErr, tc.ExpectedErr)
+			} else {
+				require.NoError(t, actualErr)
+			}
+		})
+	}
+
+	t.Run("Go", func(t *testing.T) {
+		t.Run("Plain", func(t *testing.T) {
+			validate(t, &testCase{
+				Name: "Well-formed",
+
+				RepositoryPath: filepath.Join("golang", "plain"),
+				Language:       &golang.Language{},
+			})
+		})
+		t.Run("Light", func(t *testing.T) {
+			validate(t, &testCase{
+				Name: "Repository with test files",
+
+				Before: func(repositoryPath string) {
+					// Create test files to trigger an error.
+					file, err := os.Create(filepath.Join(repositoryPath, "foo_test.go"))
+					require.NoError(t, err)
+					defer file.Close()
+					file, err = os.Create(filepath.Join(repositoryPath, "bar_test.go"))
+					require.NoError(t, err)
+					defer file.Close()
```

Comment:
Same suggestions as before. Please only create as many files as you need to trigger an error. And don't use `foo` and `bar` but what the entity actually is: `somefile`, `somepackage`, etc.